src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -525,6 +530,11 @@ public interface IStartConfiguration
             /// No update checks or analytics collection should be done.
             /// </summary>
             bool IsHeadless { get; set; }
+
+            /// <summary>
+            /// Configuration option to start Dynamo in offline mode.
+            /// </summary>
+            bool NoNetworkMode { get; set; }
```

Comment:
This allows host integrators to control this flag in their custom startup configurations or by setting it while creating a `DefaultStartConfiguration`.

Reply:
I wonder if this is a good time to do this TODO on line 512
I was mentioning this to @QilongTang today on slack.

`// TODO_Dynamo3.0: Replace the IStartConfiguration with a class or struct instance in order to avoid future breaking changes for every new option added.`

We could also look at interfaces with defaults - though I am not sure we can use default properties...

Reply:
I've made this a default property as suggested. Hopefully, it shouldn't be a breaking change any longer.

Reply:
they're not as straight forward to use as I thought... the objects only seem to inherit the default implementations when they are declared as those interfaces.

Reply:
You mean code using `RevitStartConfiguration`, for example, will need to be recompiled? I'm not sure I follow what you're suggesting we do here.